packages/lib/event-types/transformers/api-request.ts

```diff
@@ -0,0 +1,204 @@
+        return { type: integrationLabel };
+      case "phone":
+        return {
+          type: "userPhone",
+          hostPhoneNumber: location.phone,
+          displayLocationPublicly: location.public,
+        };
+      default:
+        throw new Error(`Unsupported location type '${type}'`);
+    }
+  });
+}
+
+const integrationsMappingSchema = {
+  "cal-video": z.literal("integrations:daily"),
+};
+
+const InPersonSchema = z.object({
+  type: z.literal("inPerson"),
+  address: z.string(),
+  displayLocationPublicly: z.boolean().default(false),
+});
+
+const LinkSchema = z.object({
+  type: z.literal("link"),
+  link: z.string().url(),
+  displayLocationPublicly: z.boolean().default(false),
+});
+
+const IntegrationSchema = z.object({
+  type: z.union([integrationsMappingSchema["cal-video"], integrationsMappingSchema["cal-video"]]),
+});
+
+const UserPhoneSchema = z.object({
+  type: z.literal("userPhone"),
+  hostPhoneNumber: z.string(),
+  displayLocationPublicly: z.boolean().default(false),
+});
+
+const TransformedLocationSchema = z.union([InPersonSchema, LinkSchema, IntegrationSchema, UserPhoneSchema]);
+export const TransformedLocationsSchema = z.array(TransformedLocationSchema);
+
+function transformApiEventTypeBookingFields(
+  inputBookingFields: CreateEventTypeInput_2024_06_14["bookingFields"]
+) {
+  if (!inputBookingFields) {
+    return [];
+  }
+
+  const customBookingFields = inputBookingFields.map((field) => {
+    const commonFields: CommonField = {
+      name: field.slug,
+      type: field.type,
+      label: field.label,
+      sources: [
+        {
+          id: "user",
+          type: "user",
+          label: "User",
+          fieldRequired: true,
+        },
+      ],
+      editable: "user",
+      required: field.required,
+      placeholder: "placeholder" in field && field.placeholder ? field.placeholder : "",
+    };
+
+    const options = "options" in field && field.options ? transformSelectOptions(field.options) : undefined;
+
+    if (!options) {
+      return commonFields;
+    }
+
+    return {
+      ...commonFields,
+      options,
+    };
+  });
+
+  return customBookingFields;
+}
+
+export function transformSelectOptions(options: string[]) {
+  return options.map((option) => ({
+    label: option,
+    value: option,
+  }));
+}
+
+const FieldTypeEnum = z.enum([
+  "number",
+  "boolean",
+  "address",
+  "name",
+  "text",
+  "textarea",
+  "email",
+  "phone",
+  "multiemail",
+  "select",
+  "multiselect",
+  "checkbox",
+  "radio",
+  "radioInput",
+]);
+
+const CommonFieldsSchema = z.object({
+  name: z.string(),
+  type: FieldTypeEnum,
+  label: z.string(),
+  sources: z.array(
+    z.object({
+      id: z.literal("user"),
+      type: z.literal("user"),
+      label: z.literal("User"),
+      fieldRequired: z.literal(true),
+    })
+  ),
+  editable: z.literal("user"),
+  required: z.boolean(),
+  placeholder: z.string().optional(),
+});
+
+const SystemFieldsSchema = z.object({
```

Comment:
what does System mean ?

Reply:
If you see file [transformApiEventTypeForAtom.ts](https://github.com/calcom/cal.com/pull/15457/files#diff-160b5b148862bf20a83a1a27b77be36bfa4339a5972ecb1bff22bdf83323c2cd) and then function [getBookingFields](https://github.com/calcom/cal.com/pull/15457/files#diff-160b5b148862bf20a83a1a27b77be36bfa4339a5972ecb1bff22bdf83323c2cdR127), then you can see that system fields are the default booking fields added by us.

The logic is taken from "packages/features/bookings/lib/getBookingFields.ts" that takes event-type booking fields and using "ensureBookingInputsHaveSystemFields" function adds default fields.